js/trbinance.js

```diff
@@ -0,0 +1,1334 @@
+        //                        "spotTradingEnable":1,
+        //                        "marginTradingEnable":0,
+        //                        "permissions":["SPOT"]
+        //                    },
+        //                ]
+        //            }
+        //        }
+        //
+        const result = [];
+        const data = this.safeValue (response, 'data', {});
+        const list = this.safeValue (data, 'list', []);
+        for (let i = 0; i < list.length; i++) {
+            const market = list[i];
+            // const type = this.safeInteger(market, 'type'); // Important to see endpoint for this market
+            const id = this.safeString (market, 'symbol');
+            const filters = this.safeValue (market, 'filters', []);
+            const filtersByType = this.indexBy (filters, 'filterType');
+            let minPrice = undefined;
+            let maxPrice = undefined;
+            let minQty = undefined;
+            let maxQty = undefined;
+            if ('PRICE_FILTER' in filtersByType) {
+                const filter = this.safeValue (filtersByType, 'PRICE_FILTER', {});
+                minPrice = this.safeValue (filter, 'minPrice');
+                maxPrice = this.safeValue (filter, 'maxPrice');
+            }
+            if ('LOT_SIZE' in filtersByType) {
+                const filter = this.safeValue (filtersByType, 'LOT_SIZE', {});
+                minQty = this.safeValue (filter, 'minQty');
+                maxQty = this.safeValue (filter, 'maxQty');
+            }
+            const permissions = this.safeValue (market, 'permissions');
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            let spotPermission = false;
+            let marginPermission = false;
+            for (let j = 0; j < permissions.length; j++) {
+                if (permissions[j] === 'SPOT') {
+                    spotPermission = true;
+                } else if (permissions[j] === 'MARGIN') {
+                    marginPermission = true;
+                }
+            }
+            let marginEnable = undefined;
+            let spotEnable = undefined;
+            if (this.safeInteger (market, 'marginTradingEnable') === 1) {
+                marginEnable = true;
+            } else {
+                marginEnable = false;
+            }
+            if (this.safeInteger (market, 'spotTradingEnable') === 1) {
+                spotEnable = true;
+            } else {
+                spotEnable = false;
+            }
+            const symbol = base + '/' + quote;
+            const precision = {
+                'price': this.safeInteger (market, 'quotePrecision'),
+                'amount': this.safeInteger (market, 'basePrecision'),
+            };
+            result.push (this.extend (this.fees['trading'], {
```

Comment:
Let's remove `this.extend (this.fees['trading'], ` from here. It will be extended with fees later in the base class.